test/zhr_devs/submissions/commands/download_task_test.exs

```diff
@@ -19,8 +19,9 @@ defmodule ZhrDevs.Submissions.Commands.DownloadTaskTest do
   alias ZhrDevs.Submissions.ReadModels.TaskDownloads
 
   alias Uptight.Text, as: T
+  alias Uptight.Text.Urlencoded
 
-  @task_id T.new!("onthemap-elixir-algae-witchcraft-uptight")
+  @task_id T.new!("onthemap-elixir-algae-witchcraft-uptight") |> Urlencoded.new!()
```

Comment:
Actually, we don't need the dash encoding anymore! We can literally take an object of type

```
task_name :: Text.t()
programming_language :: Language.t()
library_stack :: Library.t()
integrations :: Integration.t()
```
(feel free to alias all of these to `Text` or feel free to make them wrappers around `Text`, I think wrappers are better) 

And then encode this object simply with Jason. Get Text.t() out of encoding, and then, when you need to send this into browser or from the browser, say, to capture URL fragment, you'll do encoding / decoding then!

The only issue is that you may lose the type information, but I'd rather encode / decode everything loselessly.

If you do loseless encodings, in the end, you do **way less work** actually, so wrapping stuff into data types like `Library` may feel like a chore, but it pays off **big time** in the long run!